src/main/java/org/springframework/data/couchbase/core/query/WithConsistency.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.couchbase.core.query;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.data.annotation.QueryAnnotation;
+
+import com.couchbase.client.java.query.consistency.ScanConsistency;
+
+/**
+ * Annotation to set the scan consistency of N1QL queries with Couchbase.
+ * This controls whether couchbase waits for all changes to be processed by an index
+ * or whether stale results are acceptable.
+ * <p/>
+ * If not set, the {@link Consistency#DEFAULT_CONSISTENCY default consistency} is used.
```

Comment:
Doesn't the actual value used come from `getCouchbaseOperations().getDefaultConsistency()` which may be different than `Consistency.DEFAULT_CONSISTENCY` (`READ_YOUR_OWN_WRITES` / `STATEMENT_PLUS`)?

Reply:
Good point, I pointed the javadoc to `AbstractCouchbaseConfiguration#getDefaultConsistency()`